documentation/projects/proposals/trust_and_safety/content_report_moderation/20240122-implementation_plan_bulk_moderation_actions.md

```diff
@@ -0,0 +1,526 @@
+
+Identical to the other bulk deindex, except the action type is
+`deindexed_copyright`.
+
+### Reverse Mark Sensitive: undo marking the selected works as sensitive
+
+This action takes a queryset of `SensitiveMedia` records, rather than
+`AbstractMedia`. Each `SensitiveMedia` record has a related `media_obj`, which
+are what will be added to the ModerationDecision. It creates a
+ModerationDecision with type `reversed_mark_sensitive`, and adds the related
+media. Then it first calls `_update_es` on the `SensitiveMedia` records to
+update the elasticsearch index with the "sensitive" field for each media record,
+and finally deletes the actual `SensitiveMedia` records.
+
+Although the `SensitiveMedia` records are deleted, the ModerationDecisions
+preserve the history of these actions.
+
+### Reverse Deindex: undo deindexing the selected works
+
+This action takes a queryset of `DeletedMedia` records, rather than
+`AbstractMedia`. Like `SensitiveMedia`, each `DeletedMedia` record has a related
+`media_obj`, which will be added to the ModerationDecision. The action creates a
+ModerationDecision with type `reversed_deindex` and adds the related records.
+Then it deletes the `DeletedMedia` records.
+
+It is extremely important to note that this action does not immediately result
+in deindexed records being added back to the DB and elasticsearch indices. When
+the records were originally deindexed, they were deleted from both, and there is
+no quick way to restore them without running a data refresh. The confirmation
+page should contain a highly visible `info` block explaining this to the user.
+
+Although the `DeletedMedia` records are deleted, the ModerationDecisions
+preserve the history of these actions.
+
+### Other actions
+
+We do _not_ need to add any new action types for `rejected_reports` and
+`duplicate_reports`, which are only applicable to the Report view flow described
+in the related IP.
+
+## Views
+
+We will add one new read-only view for viewing ModerationDecisions, but most
+changes will be to existing views.
+
+### Media views
+
+We will add the custom bulk actions for `deindex_sensitive`,
+`deindex_copyright`, and `mark_sensitive` to the dropdown in the Media views.
+Bulk moderation decisions of these types will be created by taking these
+actions.
+
+However the current Media views only allow filtering records by uuid. To
+facilitate selecting records for bulk action, we'll update the `search_fields`
+for the main searchbar to search text fields (title, description, tags). We'll
+also add some additional text filters to the sidebar:
+
+- `uuid` - it may still be necessary to search by uuid, so we should preserve
+  this functionality. We include it in a separate filter (rather than keeping it
+  in `search_fields`) for search efficiency.
+- `provider`
+- `creator` - this field should have helptext reminding the user to also filter
+  by provider in order to disambiguate a specific creator.
```

Comment:
I think elasticsearch for sure -- there should be no reason we can't, since the text filters will have to be custom anyway. I'll add that.

Also, this makes me realize: we should probably leave the `search_fields` to just the id, and move the search terms for description/tags/title to the sidebar as well. I think that might be easier when customizing the implementation to query elasticsearch, and generally clearer. I'll update!